packages/elements/src/internals/machines/form/form.context.ts

```diff
@@ -29,8 +29,15 @@ export const fieldValueSelector = (name: string | undefined) => (state: Snapshot
 export const fieldHasValueSelector = (name: string | undefined) => (state: SnapshotState) =>
   Boolean(fieldValueSelector(name)(state));
 
+type MapValue<A> = A extends Map<any, infer V> ? V : never;
+
 /**
  * Selects field-specific feedback, if they exist
  */
-export const fieldFeedbackSelector = (name: string | undefined) => (state: SnapshotState) =>
-  name ? state.context.fields.get(name)?.feedback : undefined;
+// We declare an explicit return type here because TypeScript's inference results in the subtype reduction of the union
```

Comment:
Let's play more into JSDoc here.

```js
/**
 * Selects field-specific feedback, if they exist.
 * 
 * @note We declare an explicit return type here because TypeScript's inference 
 *       results in the subtype reduction of the union used for 
 *       feedback. Explicitly declaring the return type allows for 
 *       all members of the union to be included in the return type.
 */
```

Reply:
Added in [48572c4](https://github.com/clerk/javascript/pull/3812/commits/48572c46cefa35730182bb60849108413a9f5bae)!